odd-platform-api/src/main/java/com/provectus/oddplatform/repository/DataEntityRepositoryImpl.java

```diff
@@ -660,8 +660,9 @@ public void setDescription(final long dataEntityId, final String description) {
 
     @Override
     public void setInternalName(final long dataEntityId, final String businessName) {
+        final String newBusinessName = businessName != null && businessName.isEmpty() ? null : businessName;
         dslContext.update(DATA_ENTITY)
-            .set(DATA_ENTITY.INTERNAL_NAME, businessName)
+            .set(DATA_ENTITY.INTERNAL_NAME, newBusinessName)
```

Comment:
This caused unpredictable sorting behaviour. Empty values had higher priority over non-empty and nulls.